src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Legacy/ParserHelpers.cs

```diff
@@ -3,104 +3,36 @@
 
 using System;
 using System.Collections.Generic;
-using System.Globalization;
-using System.Runtime.CompilerServices;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp;
 
 namespace Microsoft.AspNetCore.Razor.Language.Legacy;
 
 internal static class ParserHelpers
 {
-    public static bool IsNewLine(char value)
-    {
-        return value switch
-        {
-            '\r' => true,     // Carriage return
-            '\n' => true,     // Line feed
-            '\u0085' => true, // Next Line
-            '\u2028' => true, // Line separator
-            '\u2029' => true, // Paragraph separator
-            _ => false
-        };
-    }
-
     public static bool IsNewLine(string value)
     {
         // We want to handle both LF and CRLF regardless of the platform.
         // We explicitly check for CRLF and IsNewLine() should return true for LF.
-        return (value is [var ch] && IsNewLine(ch)) ||
+        return (value is [var ch] && SyntaxFacts.IsNewLine(ch)) ||
                value == "\r\n";
     }
 
-    public static bool IsIdentifier(string value)
-    {
-        return IsIdentifier(value, requireIdentifierStart: true);
-    }
-
     public static bool IsIdentifier(string value, bool requireIdentifierStart)
     {
-        var span = value.AsSpan();
-
-        if (span.Length == 0)
-        {
-            return false;
-        }
-
         if (requireIdentifierStart)
         {
-            if (!IsIdentifierStart(span[0]))
-            {
-                return false;
-            }
-
-            span = span[1..];
+            return SyntaxFacts.IsValidIdentifier(value);
         }
 
-        return AllIdentifierParts(span);
-
-        static bool AllIdentifierParts(ReadOnlySpan<char> span)
+        if (value.Length == 0)
         {
-            foreach (var ch in span)
-            {
-                if (!IsIdentifierPart(ch))
-                {
-                    return false;
-                }
-            }
-
-            return true;
+            return false;
         }
-    }
 
-    public static bool IsIdentifierStart(char value)
-    {
-        return value == '_' || IsLetter(value);
+        return value.All(SyntaxFacts.IsIdentifierPartCharacter);
     }
 
-    public static bool IsIdentifierPart(char value)
-    {
-        return IsLetter(value) ||
-               IsDecimalDigit(value) ||
-              (CharUnicodeInfo.GetUnicodeCategory(value) is UnicodeCategory.Format
-                                                         or UnicodeCategory.SpacingCombiningMark
-                                                         or UnicodeCategory.NonSpacingMark
-                                                         or UnicodeCategory.ConnectorPunctuation);
-    }
-
-    public static bool IsWhitespace(char value)
-    {
-        return value == ' ' ||
-               value == '\f' ||
-               value == '\t' ||
-               value == '\u000B' || // Vertical Tab
-               char.IsSeparator(value);
```

Comment:
An important note: calling `char.IsSeparator` here was a perf optimization done in [this commit](https://github.com/dotnet/razor/commit/3c51467114d01e4f06faade4a243e13ef318678b#diff-6d835317be658da9a4706b06ad45a9d940b11c4a86d4fd2996a5642b4bedd548R70). Well meaning, but it is incorrect, as `char.IsSeparator` considers more separator categories than either C# or HTML allows for whitespace. At some point we may need to rationalize how whitespace in HTML is handed; the Razor compiler accepts all C# whitespace as whitespace, but this is a broader set of characters than is allowed in the HTML specification.